dts/bindings/pinctrl/hpmicro,hpm-pinctrl.yaml

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2022 HPMicro
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The HPMicro pin controller is a singleton node responsible for
+  controlling pin function selection and pin properties. For example, you can
+  use this node to route UART0 RX to pin PA10 and enable the pull-up resistor
+  on the pin. Remapping is also supported.
+
+  The node has the 'pinctrl' node label set in your SoC's devicetree,
+  so you can modify it like this:
+
+    &pinctrl {
+            /* your modifications go here */
+    };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+  &pinctrl {
+    pinmux_uart0: pinmux_uart0 {
+      group0 {
+        pinmux = <HPMICRO_PINMUX(HPMICRO_PORTY, 7,IOC_TYPE_IOC, 0, 2)>,
+        <HPMICRO_PINMUX(HPMICRO_PORTY, 6,IOC_TYPE_IOC, 0, 2)>;
```

Comment:
suggestion here: to simplify macro usage, make this `HPMICRO_PINMUX(Y, 7, IOC, 0, 2)>`, and use e.g. `HPMICRO_PORT ## port` in the `HPMICRO_PINMUX` macro

Reply:
Done

Reply:
update docs?

Reply:
> update docs?

Is update the use of Pinmux usage to the doc？

Reply:
> update docs?

https://github.com/zephyrproject-rtos/zephyr/compare/bc99ea109892652d625fc73c316b105c12a9ef15..03c9348b8abc6a9f5f8680e4a5868bbbdea5b2f4

Reply:
this still looks outdated vs what's on board-pinctrl.dtsi

Reply:
> this still looks outdated vs what's on board-pinctrl.dtsi

Thank you for review, I have changed it.